src/test/util/setup_common.cpp

```diff
@@ -403,6 +403,15 @@ CBlock TestChainSetup::CreateBlock(const std::vector<CMutableTransaction>& txns,
 
 TestChainSetup::~TestChainSetup()
 {
+    // Allow tx index to catch up with the block index cause otherwise
+    // we might be destroying it while scheduler still has some work for it
+    // e.g. via BlockConnected signal
+    constexpr int64_t timeout_ms = 10 * 1000;
```

Comment:
nit: why don't move `timeout_ms` inside while? so far as it constexpr variable definition, the overhead is zero, but it would be defined closer to usage